### ISSUE TITLE
Mock out VCS/go get in order to improve testing speed

### ISSUE THREAD

**Issue by twexler** [gomods/athens#1368]
**Is your feature request related to a problem? Please describe.**
During office hours, we discussed testing `athens`, and it seemed like one issue is that we have to actually call out to somewhere public to run tests. This isn't ideal for several reasons.

**Describe the solution you'd like**
Ideally, this would just be a `module.Fetcher` implemented against a set of local filesystem objects

**Describe alternatives you've considered**
None considered

**Additional context**

**Comment by marwan-at-work** [comment]
@twexler can you specify which tests exactly should be using a mock? I believe we do have a mock implementation of `module.Fetcher` but we also test the live-internet-y code for good reasons to make sure things function properly. I can help you look around if you're not sure which tests would you like to mock out. Thanks!

**Comment by yolo3301** [comment]
Seems similar to https://github.com/gomods/athens/issues/1359

**Comment by twexler** [comment]
I need to rewatch yesterday's office hours to regain context, but as mentioned this came up during that meeting

**Comment by twexler** [comment]
@yolo3301 it seems fairly similar to that issue, maybe sort of a duplicate?

**Comment by arschles** [comment]
There _could_ be two levels to this. a Mock VCS could help us isolate the caller / response parsing code from the code that sets up the command and environment. the mock binary in #1359 could validate the code to set up the command/env

**Comment by arschles** [comment]
>Seems similar to #1359?

@yolocs yes, I think this is exactly the same. #1449 should fix both of these issues
